_tour/variances.md

```diff
@@ -32,110 +36,90 @@ case class Cat(name: String) extends Animal
 case class Dog(name: String) extends Animal
 ```
 
-Both `Cat` and `Dog` are subtypes of `Animal`. The Scala standard library has a generic immutable `sealed abstract class List[+A]` class, where the type parameter `A` is covariant. This means that a `List[Cat]` is a `List[Animal]`. A `List[Dog]` is also a `List[Animal]`. Intuitively, it makes sense that a list of cats and a list of dogs are each lists of animals, and you should be able to use either of them for in place of `List[Animal]`.
-
-In the following example, the method `printAnimalNames` will accept a list of animals as an argument and print their names each on a new line. If `List[A]` were not covariant, the last two method calls would not compile, which would severely limit the usefulness of the `printAnimalNames` method.
+We can say that `Cat` is a subtype of `Animal`, and that `Dog` is also a subtype of `Animal`. That means that the following is well-typed:
 
 ```scala mdoc
-def printAnimalNames(animals: List[Animal]): Unit =
-  animals.foreach {
-    animal => println(animal.name)
-  }
+  val myAnimal: Animal = Cat("felix")
+```
 
-val cats: List[Cat] = List(Cat("Whiskers"), Cat("Tom"))
-val dogs: List[Dog] = List(Dog("Fido"), Dog("Rex"))
+What about boxes? If I have a `Box[Cat]`, is it a subtype of `Box[Animal]`, like `Cat` is a subtype of `Animal`? At first sight, it looks like that may be plausible, but if we try to do that, the compiler will tell us we have an error:
 
-// prints: Whiskers, Tom
-printAnimalNames(cats)
-
-// prints: Fido, Rex
-printAnimalNames(dogs)
+```scala
+  val myCatBox: Box[Cat] = new Box[Cat](Cat("felix"))
+  val myAnimalBox: Box[Animal] = myCatBox // this doesn't compile
+  val myAnimal: Animal = myAnimalBox.content
 ```
 
-### Contravariance
+Why could this be a problem? We can get the cat from the box, and it's still an Animal, isn't it? Well, yes. But that's not all we can do. We can also replace the cat in the box with a different animal
 
-A type parameter `A` of a generic class can be made contravariant by using the annotation `-A`. This creates a subtyping relationship between the class and its type parameter that is similar, but opposite to what we get with covariance. That is, for some `class Printer[-A]`, making `A` contravariant implies that for two types `A` and `B` where `A` is a subtype of `B`, `Printer[B]` is a subtype of `Printer[A]`.
+```scala
+  mayAnimalBox.content = Dog("fido")
+```
 
-Consider the `Cat`, `Dog`, and `Animal` classes defined above for the following example:
+There now is a Dog in the Animal box. That's all fine, you can put Dogs in Animal boxes, because Dogs are Animals. But our Animal Box is a Cat Box! You can't put a Dog in a Cat box. If we could, and then try to get the cat from our catbox, it would turn out to be a dog, breaking type soundness.
```

Comment:
s/catbox/cat box/ — worth fixing because "catbox" is a litter box 💩

Reply:
even "cat box" has the potential of being read that way, of course, but oh well :-)

Reply:
😹